userspace/libsinsp/parsers.cpp

```diff
@@ -4910,6 +4916,28 @@ void sinsp_parser::parse_setresuid_exit(sinsp_evt *evt)
 	}
 }
 
+void sinsp_parser::parse_setreuid_exit(sinsp_evt *evt)
+{
+	int64_t retval;
+
+	//
+	// Extract the return value
+	//
+	retval = evt->get_param(0)->as<int64_t>();
+
+	if(retval >= 0)
+	{
+		uint32_t new_euid = evt->get_param(1)->as<uint32_t>();
+
+		if(new_euid < std::numeric_limits<uint32_t>::max())
+		{
+			if (evt->get_thread_info()) {
+				evt->get_thread_info()->set_user(new_euid);
```

Comment:
This is not related to this PR.
Looking inside `set_user` code, why are we copying all the user info inside each thread info? wouldn't it be better to just save a shared pointer to a user struct with all the info? Is there a particular reason why we are doing this? @FedeDP

Reply:
Yes that's because if a process `foo` is started as user `pippo` but then user pippo is removed from the system (while the process runs), we should still be able to tell that process `foo` has user `pippo` even if it is not existing anymore, since that is the user it was started by.

Reply:
Now that i think about it, perhaps a shared_ptr would do the trick too!

Reply:
yep the user info should be available until the last thread using it is dead